ui/src/app/pages/analysis/interactive-notebook.tsx

```diff
@@ -149,28 +149,39 @@ export const InteractiveNotebook = fp.flow(
 
     componentDidMount(): void {
       this.props.hideSpinner();
-    }
 
-    componentDidUpdate(prevProps: Readonly<Props>) {
       const {urlParams: {ns, wsid, nbName}} = this.props;
+      if (ns && wsid && nbName) {
+        this.loadNotebook();
+      }
+    }
 
-      if (!hasNewValidProps(this.props, prevProps, [p => p.urlParams.ns, p => p.urlParams.wsid, p => p.urlParams.nbName])) {
-        return;
+    componentDidUpdate(prevProps: Readonly<Props>) {
+      if (hasNewValidProps(this.props, prevProps, [p => p.urlParams.ns, p => p.urlParams.wsid, p => p.urlParams.nbName])) {
+        this.loadNotebook();
       }
+    }
+
+    componentWillUnmount(): void {
+      this.pollAborter.abort();
+    }
 
-      workspacesApi().readOnlyNotebook(ns, wsid, nbName).then(html => {
-        this.setState({html: html.html});
-      }).catch((e) => {
+    async loadNotebook() {
+      const {ns, wsid, nbName} = this.props.urlParams;
+      try {
+        const {html} = await workspacesApi().readOnlyNotebook(ns, wsid, nbName);
+        await this.setState({html: html});
```

Comment:
why is this await necessary?

Reply:
oh, actually, probably not. setstate technically async but we don't need to wait on it to return from the function.